### PR TITLE
merge: Omit generated source columns by default

### PR DESCRIPTION
**Pull request by tsibley**
Removes default naming template and requires users to explicitly provide their own template to include source columns.  This makes the output from an `augur merge` invocation more self-documenting without columns "magically" appearing.  In the expected context of usage within a workflow, the burden of the extra option is negligible.  See also discussion on a prior PR.¹

¹ <https://github.com/nextstrain/augur/pull/1625#issuecomment-2334501940>

## Checklist

- [x] Automated checks pass
- [x] [Check][1] if you need to add a changelog message
- [x] [Check][2] if you need to add tests
- [x] [Check][3] if you need to update docs

[1]: https://github.com/nextstrain/augur/blob/-/docs/contribute/DEV_DOCS.md#updating-the-changelog
[2]: https://github.com/nextstrain/augur/blob/-/docs/contribute/DEV_DOCS.md#testing
[3]: https://github.com/nextstrain/augur/blob/-/docs/contribute/DEV_DOCS.md#when-to-update

### CODE REVIEW

**Comment by tsibley** [comment]
Once this is merged, I'd like to cut a new (major) Augur release and get it published everywhere so I can then merge https://github.com/nextstrain/measles/pull/52.

File: augur/merge.py
Comment (victorlin) [inline augur/merge.py:96]: Why keep `--no-source-columns` around when it is already the default behavior? It seems unnecessary, with the equivalent being options such as `--not-quiet` or `--no-output-metadata`.

**Comment by victorlin** [comment]
> Once this is merged, I'd like to cut a new (major) Augur release and get it published everywhere

If you end up doing this while I'm gone on lab retreat, could you also merge other breaking changes https://github.com/nextstrain/augur/pull/1613 and https://github.com/nextstrain/augur/pull/1629? Those have been approved and I just haven't gotten around to doing the actual release. The only thing is there will be changelog merge conflicts.

**Review by victorlin — APPROVED** [review]
Looks good, I'll defer to you for https://github.com/nextstrain/augur/pull/1632#discussion_r1761826666 so consider that non-blocking.

File: augur/merge.py
Comment (tsibley) [reply augur/merge.py:96]: It's a bit arbitrary/preference, but in general `--x` / `--no-x` is a nice (and very common) pattern to allow disabling previously-set options. This allows more flexibility in command invocation construction and also provides a way for invocations to opt-into usages which aren't broken by changes to the defaults. We haven't generally done this as a matter of course in Augur, but I don't think it'd be a bad idea to start and it was easy to keep this pattern here.

Comment (tsibley) [reply augur/merge.py:96]: See also [click's implementation of this pattern](https://click.palletsprojects.com/en/8.1.x/options/#boolean-flags).

Comment (victorlin) [reply augur/merge.py:96]: Thanks for the explanation, makes sense!

**Pull request merged**
